File: app/src/main/java/com/designpatterns/demo/behavioral/strategy/java/Price.java
package com.designpatterns.demo.behavioral.strategy.java;

/**
 * 价钱类 用来操作策略的上下文环境
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/5/25 上午1:03
 * @project_name DesignPatternsDemo
 */
public class Price {
    private Member member;

    public Price(Member member) {
        this.member = member;
    }

    public double quote(double booksPrice){
        return this.member.calcPrice(booksPrice);
    }
}

File: app/src/main/java/com/designpatterns/demo/behavioral/strategy/kotlin/Price.kt
package com.designpatterns.demo.behavioral.strategy.kotlin

/**
 * 价钱类 用来操作策略的上下文环境
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/5/31 上午9:59
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */
class Price(val member:Member){
    fun quote(booksPrice: Double): Double {
        return this.member.calcPrice(booksPrice)
    }
}